packages/devtools/lib/src/timeline/cpu_profile_protocol.dart

```diff
@@ -98,3 +99,107 @@ class CpuProfileProtocol {
     return name;
   }
 }
+
+/// Protocol for processing a [CpuStackFrame] and converting it to a bottom-up
+/// representation of the CPU profile.
+class BottomUpProfileProtocol {
```

Comment:
This seems more like an algorithm than a protocol.
Maybe call this
BottomUpProfileProcessor 
or 
BottomUpProfileGenerator?
Protocol makes me think this is the way a bottom up profile is sent over the wire.

Reply:
done.